prow/bugzilla/client_test.go

```diff
@@ -930,3 +932,95 @@ func TestGetExternalBugPRsOnBug(t *testing.T) {
 		})
 	}
 }
+
+func TestGetAllClones(t *testing.T) {
+	fake := &Fake{}
+	fake.Bugs = map[int]Bug{}
+	fake.BugComments = map[int][]Comment{}
+	bug1Create := &BugCreate{
+		Summary: "Dummy bug to test getAllClones",
+	}
+	bug1ID, err := fake.CreateBug(bug1Create)
+	if err != nil {
+		t.Fatalf("Error while creating bug in Fake!\n")
+	}
+	bug1, err := fake.GetBug(bug1ID)
+	bug2ID, err := fake.CloneBug(bug1)
+	bug2, err := fake.GetBug(bug2ID)
+	bug3ID, err := fake.CloneBug(bug2)
+	bug1, err = fake.GetBug(bug1ID)
+	bug2, err = fake.GetBug(bug2ID)
+	bug3, err := fake.GetBug(bug3ID)
+	testcases := map[int]struct {
+		bug            *Bug
+		expectedClones sets.Int
+	}{
+		bug1ID: {
+			bug1,
+			sets.NewInt(bug2ID, bug3ID),
+		},
+		bug2ID: {
+			bug2,
+			sets.NewInt(bug1ID, bug3ID),
+		},
+		bug3ID: {
+			bug3,
+			sets.NewInt(bug1ID, bug2ID),
+		},
+	}
+	for tc, tp := range testcases {
+		t.Run(strconv.Itoa(tc), func(t *testing.T) {
+			clones, err := getAllClones(fake, tp.bug)
+			if err != nil {
+				t.Errorf("Error occurred when none was expected: %v", err)
+			}
+			if len(tp.expectedClones) != len(clones) {
+				t.Errorf("Mismatch in number of clones - expected: %d, got %d", len(tp.expectedClones), len(clones))
+			}
+			for _, clone := range clones {
+				if ok := tp.expectedClones.Has(clone.ID); !ok {
+					t.Errorf("Unexpected clone found in list - expecting: %v, got: %d", tp.expectedClones, clone.ID)
+				}
+			}
+		})
+
+	}
+
+}
+
+func TestGetRootForClone(t *testing.T) {
+	fake := &Fake{}
+	fake.Bugs = map[int]Bug{}
+	fake.BugComments = map[int][]Comment{}
+	bug1Create := &BugCreate{
+		Summary: "Dummy bug to test getAllClones",
+	}
+	bug1ID, err := fake.CreateBug(bug1Create)
+	if err != nil {
+		t.Fatalf("Error while creating bug in Fake!\n")
+	}
+	bug1, err := fake.GetBug(bug1ID)
+	bug2ID, err := fake.CloneBug(bug1)
+	bug2, err := fake.GetBug(bug2ID)
+	bug3ID, err := fake.CloneBug(bug2)
+	bug1, err = fake.GetBug(bug1ID)
+	bug2, err = fake.GetBug(bug2ID)
+	bug3, err := fake.GetBug(bug3ID)
+	testcases := map[int]*Bug{
```

Comment:
Instead of defining the testcases as maps, could you please define them as a
`[]struct` that contains all necesary parameters and a descriptive name?

See for example:
* https://github.com/openshift/ci-tools/blob/cb0cd9370002db5f9cd474d83a6e8bae27663ebc/pkg/diffs/diffs_test.go#L205
* https://github.com/openshift/ci-tools/blob/cb0cd9370002db5f9cd474d83a6e8bae27663ebc/pkg/diffs/diffs_test.go#L386

Reply:
Ok... I have modified the testcases to be structs instead of maps with a relevant name.